### PR TITLE
Add FromStr for FixedOffset

### LINKED ISSUE #314: Add a FixedOffset parser
In [my project](https://gitlab.com/xmpp-rs/xmpp-parsers), an [XMPP](https://xmpp.org) element parser, I need to be able to parse a timezone on its own, in order to implement [XEP-0202](https://xmpp.org/extensions/xep-0202.html).

I’m currently using [a super ugly workaround](https://gitlab.com/xmpp-rs/xmpp-parsers/blob/master/src/time.rs#L46-49) by creating a DateTime from a random date concatenated with the timezone I want to parse, and then extracting the timezone from it.

A correct way to implement that could be to `impl std::str::FromStr for FixedOffset`, which would return only the timezone information and reject any other string.

### PR DESCRIPTION
**Pull request by yshui**
Closes #314

### CODE REVIEW

File: src/format/scan.rs [hunk 1]
```diff
@@ -221,7 +221,7 @@ pub(super) fn trim1(s: &str) -> &str {
 
 /// Consumes one colon char `:` if it is at the front of `s`.
 /// Always returns `Ok(s)`.
-pub(super) fn consume_colon_maybe(mut s: &str) -> ParseResult<&str> {
+pub(crate) fn consume_colon_maybe(mut s: &str) -> ParseResult<&str> {
```
Comment (pitdicker) [inline src/format/scan.rs:224]: These two methods don't need to be public for the crate.

But maybe you should keep them like this, and put the `FromStr` implementation in `offset/fixed.rs`. It seems to be an exception to put them in this file.

**Review by pitdicker — COMMENTED** [review]
Seems like a sensible addition to me.

File: src/format/mod.rs [hunk 2]
```diff
@@ -991,6 +992,15 @@ impl FromStr for Weekday {
     }
 }
 
+/// Parsing a `str` into a `FixedOffset` uses the format [`%z`](./format/strftime/index.html).
```
[the remark this answers is not in the document]
Comment (djc) [reply src/format/mod.rs:995]: Please move this impl below the `FixedOffset` definition.

**Review by djc — COMMENTED** [review]
Makes sense, thanks!

File: src/format/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jtmoon79) [reply src/format/mod.rs:995]: This addition should be move to `src/offset/fixed.rs`.

File: src/format/mod.rs [hunk 3]
```diff
@@ -991,6 +992,15 @@ impl FromStr for Weekday {
...
+impl FromStr for FixedOffset {
+    type Err = ParseError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (_, offset) = scan::timezone_offset(s, scan::consume_colon_maybe)?;
+        Self::east_opt(offset).ok_or(OUT_OF_RANGE)
+    }
+}
+
```
Comment (jtmoon79) [inline src/format/mod.rs:1003]: This addition should be move to `src/offset/fixed.rs`.

**Review by jtmoon79 — CHANGES_REQUESTED** [review]

**Comment by pitdicker** [comment]
Merged in https://github.com/chronotope/chrono/pull/1157. Thank you for working on this.

**Pull request closed without merging**